pulp_file/tests/functional/__init__.py

```diff
@@ -229,6 +229,20 @@ def file_fixture_server(file_fixtures_root, gen_fixture_server):
     return gen_fixture_server(file_fixtures_root, None)
 
 
+@pytest.fixture
+def gen_file_remote(file_fixture_server):
+    def _gen_file_remote(manifest_path=None, **kwargs):
+        url = kwargs.get("url")
+        if not url:
+            assert manifest_path is not None
+            url = file_fixture_server.make_url(manifest_path)
+        data = {"url": url, "name": str(uuid.uuid4())}
+        data.update(kwargs)
+        return data
```

Comment:
It's fine to "adjust" kwargs:
```suggestion
        if not url in kwargs:
            assert manifest_path is not None
            kwargs["url"] = file_fixture_server.make_url(manifest_path)
        kwargs.set_default("name", str(uuid.uuid4()))
        return kwargs
```

Also let's move this logic into the file_remote_factory.